execution_engine/src/storage/global_state/scratch.rs

```
@@ -120,7 +131,10 @@ impl StateReader<Key, StoredValue> for ScratchGlobalStateView {
             key,
         )? {
             ReadResult::Found(value) => {
-                self.cache.write().unwrap().insert(*key, value.clone());
+                self.cache
+                    .write()
+                    .expect("Cache lock should not be poisoned")
```

We can throw an error here too...